extensions/mssql/src/utils.ts

```diff
@@ -29,6 +30,33 @@ export function getAppDataPath() {
 	}
 }
 
+/**
+ * Get a file name that is not already used in the target directory
+ * @param filePath source notebook file name
+ * @param fileExtension file type
+ */
+export function getTargetFileName(filePath: string, fileExtension: string): string {
+	const targetDirectory = os.homedir();
+	const baseName = path.basename(filePath, fileExtension);
```

Comment:
Isn't this logic going to produce `cluster-status.ipynb-1.ipynb`? I just ran into the same issue, and used the below logic that doesn't require passing in the fileExtension, and would output `cluster-status-1.ipynb`. I recommend using the same here. Ideally we'd share the code but as this is in separate extensions, I guess we'll have to dupe for now.
https://github.com/microsoft/azuredatastudio/pull/6093/files

```ts
	private getUntitledUri(originalTitle: string): vscode.Uri {
		let title = originalTitle;
		let nextVal = 0;
		let ext = path.extname(title);
		while (!isEditorTitleFree(title)) {
			if (ext) {
				// Need it to be `Readme-0.txt` not `Readme.txt-0`
				let titleStart = originalTitle.slice(0, originalTitle.length - ext.length);
				title = `${titleStart}-${nextVal}${ext}`;
			} else {
				title = `${originalTitle}-${nextVal}`;
			}
			nextVal++;
		}
		return vscode.Uri.parse(`untitled:${title}`);
	}

... in a utils module:
export function isEditorTitleFree(title: string): boolean {
	let hasTextDoc = vscode.workspace.textDocuments.findIndex(doc => doc.isUntitled && doc.fileName === title) > -1;
	let hasNotebookDoc = azdata.nb.notebookDocuments.findIndex(doc => doc.isUntitled && doc.fileName === title) > -1;
	return !hasTextDoc && !hasNotebookDoc;
} 

```

Reply:
path.basename only get's the filename without the extension. So it only get's cluster-status as the basename and extension '.ipynb' is added when forming the targetFileName. I tested this and it works. Do you still want me to change it and use your implementation for uniformity?

Reply:
I'd prefer if you can skip passing in '.ipynb' as a hardcoded thing. Looking at path.basename, it's working as you pass in the extension and it knows to skip that part. Could you try:
```ts
	let fileExtension = path.extname(filePath); 
       const baseName = path.basename(filePath, fileExtension);
```
So that we skip this? In the future, it'd be nice that this can work for any file.